src/lib.rs

```diff
@@ -465,41 +462,132 @@ struct CLruNode<K, V> {
     value: V,
 }
 
-/// An LRU cache with constant time operations.
-#[derive(Debug)]
-pub struct CLruCache<K, V, S = RandomState> {
-    lookup: HashMap<Key<K>, usize, S>,
-    storage: FixedSizeList<CLruNode<K, V>>,
+/// Trait used to retrieve the weight of a value.
+pub trait WeightScale<V> {
```

Comment:
At first sight, I think this merits a refactoring: Putting these structs / traits into their own files.

For already existing code / types, I would address that in a different PR. But for new types, probably the best time is when introducing them.

Reply:
Fixed